### ISSUE TITLE
[BUG] Install goes into an infinite loop when there's a peer dependency conflict

### ISSUE THREAD

**Issue by rafgraph** [npm/cli#1945]
### Current Behavior:
`npm i --save-dev microbundle` freezes (infinite loop) at:
`⸨      ░░░░░░░░░░░░⸩ ⠧ idealTree:@rollup/plugin-json: timing idealTree:node_mod`

This is probably related to this `microbundle` peer dependency issue, https://github.com/developit/microbundle/issues/733, but npm should error not freeze.

Note that the freeze is flaky, in some of my repos it freezes and some it errors (not sure why), but when creating a new blank project it freezes so I thought I should report it. For a repo that it errors in (and doesn't freeze), you can git clone https://github.com/rafgraph/spa-github-pages and `npm i --save-dev microbundle`.

### Expected Behavior:
For npm not to freeze and provide an error.

### Steps To Reproduce:
```
mkdir test
cd test
npm init --yes
npm i --save-dev microbundle
```

### Environment:
- OS: MacOS 10.15.7
- Node: 12.8.3
- npm: 7.0.0-rc.4

**Comment by jsg2021** [comment]
use `--loglevel silly`  ... it's probably in an infinite loop

**Comment by rafgraph** [comment]
Yup it's an infinite loop

**Comment by rafgraph** [comment]
```
...
npm sill fetch manifest rollup@^1.20.0
npm http fetch GET 304 https://registry.npmjs.org/rollup 46ms (from cache)
npm sill placeDep ROOT @types/resolve@0.0.8 Symbol(OK) for: @rollup/plugin-node-resolve@6.1.0
npm sill placeDep ROOT is-module@^1.0.0 Symbol(OK) for: @rollup/plugin-node-resolve@6.1.0
npm sill placeDep ROOT rollup@^1.20.0 Symbol(REPLACE) for: @rollup/plugin-node-resolve@6.1.0
npm timing idealTree:node_modules/@rollup/plugin-node-resolve Completed in 62ms
npm sill placeDep ROOT rollup@^2.3.4 Symbol(REPLACE) for: @rollup/plugin-commonjs@13.0.2
npm timing idealTree:node_modules/@rollup/plugin-commonjs Completed in 3ms
npm sill placeDep ROOT rollup@^1.20.0 Symbol(REPLACE) for: @rollup/plugin-node-resolve@6.1.0
npm timing idealTree:node_modules/@rollup/plugin-node-resolve Completed in 2ms
npm sill placeDep ROOT rollup@^2.3.4 Symbol(REPLACE) for: @rollup/plugin-commonjs@13.0.2

etc...
```

**Comment by rafgraph** [comment]
This appears to be solved in npm v7.0.7
